### PR TITLE
Network subsystem remove os rules

### PR DESCRIPTION
**Pull request by dorileo**

### CODE REVIEW

File: google_guest_agent/network/manager/manager_linux.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+func init() {
+	wickedCommand, err := exec.LookPath("wicked")
+	if err != nil {
+		logger.Infof("failed to find wicked path, falling back to default: %+v", err)
+		wickedCommand = defaultWickedCommand
+	}
+
+	knownNetworkManagers = []Service{
```
Comment (a-crate) [inline google_guest_agent/network/manager/manager_linux.go:30]: I think setting an already declared var in `init` to a value that's more or less static is needlessly confusing. IMO this slice should be set as part of the declaration and `init()` in `wicked_linux.go` should edit wickedCommand if exec.LookPath is successful.

File: google_guest_agent/network/manager/netplan_linux.go [hunk 2]
```diff
@@ -244,7 +268,7 @@ func (nd networkdNetplanDropin) write(n netplan, iface string) error {
 	logger.Infof("writing systemd drop in to: %s", dropinFile)
 
 	dropinDir := filepath.Dir(dropinFile)
-	err := os.MkdirAll(dropinDir, 0750)
+	err := os.MkdirAll(dropinDir, 0755)
```
Comment (a-crate) [inline google_guest_agent/network/manager/netplan_linux.go:271]: Why did this change?

File: google_guest_agent/network/manager/netplan_linux.go [hunk 3]
```diff
@@ -88,11 +79,22 @@ type netplanEthernet struct {
...
+	DHCP4Overrides *netplanDHCPOverrides `yaml:"dhcp4-overrides,omitempty"`
+
 	// DHCPv6 determines if DHCPv6 support must be enabled to such an interface.
 	DHCPv6 *bool `yaml:"dhcp6,omitempty"`
+
+	DHCP6Overrides *netplanDHCPOverrides `yaml:"dhcp6-overrides,omitempty"`
+}
+
+type netplanDHCPOverrides struct {
+	UseDomains bool `yaml:"use-domains,omitempty"`
```
Comment (a-crate) [inline google_guest_agent/network/manager/netplan_linux.go:97]: Can you add a comment about using this to add a search domain for whatever DNS domain the instance is in

**Review by a-crate — COMMENTED** [review]
lgtm

File: google_guest_agent/network/manager/netplan_linux.go (hunk 2 above)
Comment (dorileo) [reply google_guest_agent/network/manager/netplan_linux.go:271]: it was wrong in the first place.

File: google_guest_agent/network/manager/netplan_linux.go (hunk 3 above)
Comment (dorileo) [reply google_guest_agent/network/manager/netplan_linux.go:97]: done. thanks.

**Review by a-crate — APPROVED** [review]

**Pull request merged**